lab_3/main.py

```diff
@@ -222,7 +334,24 @@ def save(self, name: str) -> int:
         :param name: name of the json file with .json format
         :return: 0 if profile saves, 1 if any errors occurred
         """
-        pass
+        if not isinstance(name, str):
+            return 1
+        profile_as_dict = {}
+        freq = {}
+        word = ''
+        for instance in self.tries:
+            for n_gram, frequency in instance.n_gram_frequencies.items():
+                for integer in n_gram:
+                    word += self.storage.get_letter_by_id(integer)
+                freq[word] = frequency
+                word = ''
+        profile_as_dict['freq'] = freq
+        profile_as_dict['n_words'] = self.n_words
+        profile_as_dict['name'] = self.language
+        with open(name, 'w', encoding='UTF-8') as file:
+            json_string = json.dumps(profile_as_dict)
+            file.write(json_string)
```

Comment:
There is the `.dump` method in the `json` package that combines these two lines.

Reply:
done